FlowCrypt/Controllers/Compose/ComposeViewController.swift

```diff
@@ -211,57 +221,52 @@ extension ComposeViewController {
 
         showSpinner("sending_title".localized)
 
-        Promise<Bool> { [weak self] in
-            try await(self!.encryptAndSendMessage())
-        }.then(on: .main) { [weak self] sent in
-            if sent { // else it must have shown error to user
-                self?.handleSuccessfullySentMessage()
-            }
-        }.catch(on: .main) { [weak self] error in
-            self?.showAlert(error: error, message: "compose_error".localized)
-        }
+        encryptAndSendMessage()
     }
 
-    private func encryptAndSendMessage() -> Promise<Bool> {
-        Promise<Bool> { [weak self] () -> Bool in
-            guard let self = self else { return false }
-
-            let recipients = self.contextToSend.recipients
-
-            guard recipients.isNotEmpty else {
-                assertionFailure("Recipients should not be empty. Fail in checking")
-                return false
-            }
-
-            guard let text = self.contextToSend.message else {
-                assertionFailure("Text and Email should not be nil at this point. Fail in checking")
-                return false
-            }
-
-            let subject = self.input.subjectReplyTitle
-                ?? self.contextToSend.subject
-                ?? "(no subject)"
-
-            guard let myPubKey = self.dataService.publicKey else {
-                self.showAlert(message: "compose_no_pub_sender".localized)
-                return false
-            }
-
-            guard let allRecipientPubs = self.getPubKeys(for: recipients) else {
-                return false
-            }
-
-            let encrypted = self.encryptMsg(
-                pubkeys: allRecipientPubs + [myPubKey],
-                subject: subject,
-                message: text,
-                to: recipients.map { $0.email }
-            )
-
-            try await(self.messageSender.sendMail(mime: encrypted.mimeEncoded))
-
-            return true
+    private func encryptAndSendMessage() {
+        let recipients = self.contextToSend.recipients
+        
+        guard recipients.isNotEmpty else {
+            return
+        }
```

Comment:
Should add back error handling, not just return and ignore. If it's input validation error from user, then we render an alert and return. If it's a nilcheck that was already supposed to be checked earlier and we're just sanity-checking our code, than that's a coding error and should result in a crash, not be ignored.

Reply:
I would suggest to use implementation with `Result<SendableMessage, SendMessageError>` as it was in previous commit for all this checks instead of returning true/false, and use asynchronous Future only for send message operation.